tests/ludwig/utils/test_config_utils.py

```diff
@@ -57,3 +72,149 @@ def test_set_fixed_preprocessing_params_cache_embeddings(encoder_params: Dict[st
+def ecd_config_object_llm_encoder_multiple_features(
+    ecd_config_dict_llm_encoder_multiple_features: Dict[str, Any]
+) -> ModelConfig:
+    return ModelConfig.from_dict(ecd_config_dict_llm_encoder_multiple_features)
+
+
+@pytest.fixture()
+def ecd_config_dict_no_llm_encoder() -> Dict[str, Any]:
+    return {
+        MODEL_TYPE: MODEL_ECD,
+        INPUT_FEATURES: [{TYPE: TEXT, NAME: "in1", ENCODER: {TYPE: "parallel_cnn"}}],
+        OUTPUT_FEATURES: [{TYPE: BINARY, NAME: "out1"}],
+    }
+
+
+@pytest.fixture()
+def ecd_config_object_no_llm_encoder(ecd_config_dict_no_llm_encoder: Dict[str, Any]) -> ModelConfig:
+    return ModelConfig.from_dict(ecd_config_dict_no_llm_encoder)
+
+
+@pytest.fixture()
+def ecd_config_dict_no_text_features() -> Dict[str, Any]:
+    return {
+        MODEL_TYPE: MODEL_ECD,
+        INPUT_FEATURES: [{TYPE: BINARY, NAME: "in1"}],
+        OUTPUT_FEATURES: [{TYPE: BINARY, NAME: "out1"}],
+    }
+
+
+@pytest.fixture()
+def ecd_config_object_no_text_features(ecd_config_dict_no_text_features: Dict[str, Any]) -> ModelConfig:
+    return ModelConfig.from_dict(ecd_config_dict_no_text_features)
+
+
+@pytest.fixture()
+def gbm_config_dict_llm_encoder() -> Dict[str, Any]:
+    return {
+        MODEL_TYPE: MODEL_GBM,
+        INPUT_FEATURES: [{TYPE: TEXT, NAME: "in1", ENCODER: {TYPE: "tf_idf"}}],
+        OUTPUT_FEATURES: [{TYPE: BINARY, NAME: "out1"}],
+    }
+
+
+@pytest.fixture()
+def gbm_config_object_llm_encoder(gbm_config_dict_llm_encoder: Dict[str, Any]) -> ModelConfig:
+    return ModelConfig.from_dict(gbm_config_dict_llm_encoder)
+
+
+@pytest.mark.parametrize(
+    "config,expectation",
+    [
+        # LLM configurations
+        ("llm_config_dict", True),
+        ("llm_config_object", True),
+        # LLM encoder configurations
+        ("ecd_config_dict_llm_encoder", True),
+        ("ecd_config_object_llm_encoder", True),
+        # LLM encoder configurations, multiple features
+        ("ecd_config_dict_llm_encoder_multiple_features", True),
+        ("ecd_config_object_llm_encoder_multiple_features", True),
+        # ECD configuration with text feature and non-LLM encoder
+        ("ecd_config_dict_no_llm_encoder", False),
+        ("ecd_config_object_no_llm_encoder", False),
+        # ECD configuration with no text features
+        ("ecd_config_dict_no_text_features", False),
+        ("ecd_config_object_no_text_features", False),
+        # GBM configuration. "tf_idf" is the only valid text encoder
+        ("gbm_config_dict_llm_encoder", False),
+        ("gbm_config_object_llm_encoder", False),
+    ],
+)
+def test_is_or_uses_llm(config, expectation, request):
+    """Test LLM detection on a variety of configs. Configs that use an LLM anywhere should return True, otherwise
+    False.
+
+    Args:
+        config: The name of the config fixture to test
+        expectation: The expected result
+        request: pytest `request` fixture
+    """
+    config = request.getfixturevalue(config)
```

Comment:
@jeffkinnison Just curious - is there a reason you decided to create all of the tests as individual (and nested) fixtures as opposed to just creating each of them directly as parameterized tests?

Reply:
It was mostly for readability on my part, though it does also allow us to reuse the configs for both the dict and object cases.

Reply:
Got it! Thanks for the changes with making them module scoped. I think that because they're relatively light weight fixtures, this is okay to do and shouldn't slow things down or cause memory related issues.

Reply:
Last callout is that because the fixtures are dependent on each other, I think it may prevent pytest from parallelizing some of the tests. Again, it seems okay in this case given that the test itself is mostly an in place config or dict check and nothing too heavy, so each of the tests should run quickly.